rollout/trafficrouting/istio/istio_test.go

```diff
@@ -616,3 +617,255 @@ func TestUpdateHashDestinationRuleNotFound(t *testing.T) {
+	r := NewReconciler(ro, client, record.NewFakeEventRecorder(), vsvcLister, druleLister)
+	client.ClearActions()
+	err := r.SetWeight(10)
+	assert.NotNil(t, err)
+	assert.True(t, k8serrors.IsNotFound(err))
+}
+
+// TestReconcileAmbiguousRoutes tests when we omit route names and there are multiple routes in the VirtualService
+func TestMultipleVirtualServiceReconcileAmbiguousRoutes(t *testing.T) {
+	obj1 := unstructuredutil.StrToUnstructuredUnsafe(sampleRouteVirtualService1)
+	obj2 := unstructuredutil.StrToUnstructuredUnsafe(sampleRouteVirtualService2)
+	client := testutil.NewFakeDynamicClient(obj1, obj2)
+	multipleVirtualService := []v1alpha1.IstioVirtualService{{Name: "vsvc1", Routes: nil}, {Name: "vsvc2", Routes: nil}}
+	ro := multiVsRollout("stable", "canary", multipleVirtualService)
+	vsvcLister, druleLister := getIstioListers(client)
+	r := NewReconciler(ro, client, record.NewFakeEventRecorder(), vsvcLister, druleLister)
+	client.ClearActions()
+	err := r.SetWeight(0)
+	assert.Equal(t, "VirtualService spec.http[] must have exactly one route when omitting spec.strategy.canary.trafficRouting.istio.virtualService.routes", err.Error())
+}
+
+// TestReconcileInferredSingleRoute we can support case where we infer the only route in the VirtualService
+func TestMultipleVirtualServiceReconcileInferredSingleRoute(t *testing.T) {
+	obj1 := unstructuredutil.StrToUnstructuredUnsafe(singleRouteMultipleVirtualService1)
+	obj2 := unstructuredutil.StrToUnstructuredUnsafe(singleRouteMultipleVirtualService2)
+	client := testutil.NewFakeDynamicClient(obj1, obj2)
+	multipleVirtualService := []v1alpha1.IstioVirtualService{{Name: "vsvc1", Routes: nil}, {Name: "vsvc2", Routes: nil}}
+	ro := multiVsRollout("stable", "canary", multipleVirtualService)
+	vsvcLister, druleLister := getIstioListers(client)
+	r := NewReconciler(ro, client, record.NewFakeEventRecorder(), vsvcLister, druleLister)
+	client.ClearActions()
+	err := r.SetWeight(10)
+	assert.NoError(t, err)
+	actions := client.Actions()
+	assert.Len(t, actions, 2)
+	assert.Equal(t, "update", actions[0].GetVerb())
+	assert.Equal(t, "update", actions[1].GetVerb())
+
+	// Verify we actually made the correct change
+	vsvcUn1, err := client.Resource(istioutil.GetIstioVirtualServiceGVR()).Namespace(ro.Namespace).Get(context.TODO(), "vsvc1", metav1.GetOptions{})
+	assert.NoError(t, err)
+	routes1, _, _ := unstructured.NestedSlice(vsvcUn1.Object, "spec", "http")
+	route1 := routes1[0].(map[string]interface{})
+	checkDestination(t, route1, "stable", 90)
+	checkDestination(t, route1, "canary", 10)
+
+	vsvcUn2, err := client.Resource(istioutil.GetIstioVirtualServiceGVR()).Namespace(ro.Namespace).Get(context.TODO(), "vsvc2", metav1.GetOptions{})
+	assert.NoError(t, err)
+	routes2, _, _ := unstructured.NestedSlice(vsvcUn2.Object, "spec", "http")
+	route2 := routes2[0].(map[string]interface{})
+	checkDestination(t, route2, "stable", 90)
+	checkDestination(t, route2, "canary", 10)
```

Comment:
The above check can be done in a for  look to reduce code duplication

```go
for _, vsvName := range []string{"vsvc1", "vsvc2"} {
        vsvcUn, err := client.Resource(istioutil.GetIstioVirtualServiceGVR()).Namespace(ro.Namespace).Get(context.TODO(), vsvName, metav1.GetOptions{})
	assert.NoError(t, err)
	routes1, _, _ := unstructured.NestedSlice(vsvcUn.Object, "spec", "http")
	route1 := routes1[0].(map[string]interface{})
	checkDestination(t, route1, "stable", 90)
	checkDestination(t, route1, "canary", 10)
}
```

Reply:
Thanks!

Reply:
Hi,  @darshanhs09, please leave it unresolved as you have not made the change as request.

Reply:
Hi @huikang, I missed to update this change. Sorry for that.
Please check the latest.